### DISCUSSION TITLE
There’s a problem with my search function on my GitHub site.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#58766] [Programming Help]
### Body

function search(){
  var searchInput = document.getElementById("search");
  var searchValue = searchInput.value;
  if(!searchValue){//searchValue is falsy
    var searchNotFound = document.createElement("h1");
    searchNotFound.innerHTML = "<h1>JavaScript Error: searchValue not found! <a href='https://pytutorials5.wordpress.com/2023/05/14/hello-world/'>Report bug</a></h1>";
    searchInput.style.display = "none";
    searchNotFound.onclick = searchInput.style.display = "block";
    document.body.appendChild(searchNotFound);
    alert("Search value is " + searchValue);
    return true;
  }
  if(searchValue){
    if(searchValue != ""){
      window.location.assign("https://pytutorials.github.io/" + searchValue);
      return false;//SearchValue is not valid
    }
    if(searchValue == null || searchValue == ""){
      var h1 = document.createElement("h1");
      h1.innerText = "Enter a page!");
      h1.style.position = "absolute";
      h1.style.bottom = "100%";
      var br = document.createElement("br");
      var em = document.createElement("em");
      em.innerText = "Thanks for using PyTutorials!";
      em.style.position = "absolute";
      em.style.top = "100%";
      em.style.right = "50%";
      document.body.appendChild(h1);
      document.body.appendChild(br);
      document.body.appendChild(em);
      return true;
    }
  }
}

Home page(Written in Markdown)
# PyTutorials - Home
  <p>print("Hello World!")</p>
  <a href="index">Home</a> <a>|</a> <a href="https://pytutorials.github.io/tutorials">Tutorials</a> | <a id="signup" href="signup" style="display:none;">Signup</a>
<div class="search-form">
  <form action="javascript:searchDefined();">
    <input type="text" value="" id="search" placeholder="Search for a page..."/>
    <button onlckck="searchDefined();" id="submit">Submit(may not work)</button>
    <noscript>You can't signup or search for different pages if Javascript isn't enabled.</noscript>
  </form>
 </div>
<hr/>
<script src="search.js"></script>
<script src="errorhandle.js"></script>
<script src="login.js"></script>
<script src="timesVisited.js"></script>
#### Example 1

* Output Hello World

```python

print("Hello World!") #This outputs the text Hello World!
```

#### Example 2
* Get a list of files in the current directory and output them

```python
import os
files = []
for file in os.listdir():
  if os.path.isfile(file):
   if file == "yourfile.py":
     continue
   else:
     files.append(file)
  else:
     continue
 ```
 
 <h4>Example 3</h4>
* Comments

```python
#This is a comment it is usually used to explain what your code does. The computer ignores this.
"""This is a multiline comment"""
```
For more examples click this [link](tutorials/index)

[Test your code here](https://onlinegdb.com)

[Add Recomendations](https://pytutorials5.wordpress.com/2023/05/14/hello-world/)
[Python examples](https://github.com/pytutorials/Python-Examples)
<a href="https://github.com/pytutorials/pytutorials.github.io/blob/main/README.md">About this project</a>
<p>This project is closed source. <a href="https://en.m.wikipedia.org/wiki/Comparison_of_open-source_and_closed-source_software">What is closed source?</a></p>
<footer>Written in HTML, Markdown, CSS, and JavaScript.</footer>
<a href="#top">Go back to the top of the page</a>
<footer>Make sure the url is pytutorials.github.io</footer>

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ghost** [comment]
https://github.com/pytutorials/pytutorials.github.io/edit/main/index.md

**Comment by kathlene-cory** [comment]
Hi @WorldsDumbestCoder - 

What exactly are you trying to do here? Or which part is broken that you would like to fix? 

The first potential issue I see is: 
`if(!searchValue){//searchValue is falsy`

Are to trying to check if the value is empty? Or when do you want a false condition to return?

**Reply by kathlene-cory** [reply]
Then further down in this conditional
`if(searchValue){
if(searchValue != ""){
window.location.assign("https://pytutorials.github.io/" + searchValue);
return false;//SearchValue is not valid
}`

If the searchValue is true and not empty, why should it return false?

**Reply by ghost** [reply]
I meant return false because no input is provided.
